apps/server/test/modules/task/submission.e2e-spec.ts

```diff
@@ -37,128 +37,250 @@ class API {
 			result: response.body as SubmissionStatusListResponse,
 		};
 	}
+
+	async delete(submissionId: string) {
+		const response = await request(this.app.getHttpServer())
+			.delete(`/submissions/${submissionId}`)
+			.set('Accept', 'application/json')
+			.set('Authorization', 'jwt');
+
+		return {
+			status: response.status,
+			error: response.body as ApiValidationError,
+			result: response.body as SubmissionStatusListResponse,
+		};
+	}
 }
 
 describe('Submission Controller (e2e)', () => {
-	let app: INestApplication;
-	let currentUser: ICurrentUser;
-	let api: API;
-	let em: EntityManager;
-
-	beforeAll(async () => {
-		const module: TestingModule = await Test.createTestingModule({
-			imports: [ServerTestModule],
-		})
-			.overrideGuard(JwtAuthGuard)
-			.useValue({
-				canActivate(context: ExecutionContext) {
-					const req: Request = context.switchToHttp().getRequest();
-					req.user = currentUser;
-
-					return true;
-				},
+	describe('find statuses by task', () => {
+		let app: INestApplication;
+		let currentUser: ICurrentUser;
+		let api: API;
+		let em: EntityManager;
+
+		beforeAll(async () => {
+			const module: TestingModule = await Test.createTestingModule({
+				imports: [ServerTestModule],
 			})
-			.compile();
+				.overrideGuard(JwtAuthGuard)
+				.useValue({
+					canActivate(context: ExecutionContext) {
+						const req: Request = context.switchToHttp().getRequest();
+						req.user = currentUser;
 
-		app = module.createNestApplication();
-		await app.init();
-		api = new API(app);
-		em = module.get(EntityManager);
-	});
+						return true;
+					},
+				})
+				.compile();
 
-	beforeEach(async () => {
-		await cleanupCollections(em);
-	});
+			app = module.createNestApplication();
+			await app.init();
+			api = new API(app);
+			em = module.get(EntityManager);
+		});
 
-	afterAll(async () => {
-		await app.close();
-	});
+		beforeEach(async () => {
+			await cleanupCollections(em);
+		});
 
-	describe('WHEN user is not authorized', () => {
-		it('should return 401', async () => {
-			const taskId = 'id';
+		afterAll(async () => {
+			await app.close();
+		});
 
-			const { status } = await api.findStatusesByTask(taskId);
+		describe('WHEN user is not authorized', () => {
```

Comment:
It's not your change, but this should be "user is not authenticated", because it is the 401 the CurrentUser decorator throws when there is no user.